packages/components/modals/teritoriNameService/TNSNameFinderModal.tsx

```diff
@@ -105,6 +110,19 @@ export const TNSNameFinderModal: React.FC<{
 }> = ({ visible, onClose, onEnter }) => {
   const { name, setName } = useTNS();
 
+  const selectedWallet = useSelectedWallet();
+  const [network] = parseUserId(selectedWallet?.userId);
+  const networkId = network?.id;
+  const cosmosNetwork = getCosmosNetwork(networkId);
+  const normalizedTokenId = (
+    name + cosmosNetwork?.nameServiceTLD || ""
+  ).toLowerCase();
+
+  const tokenId = name + cosmosNetwork?.nameServiceTLD || "";
+
+  const { nsMintPrice: price } = useNSMintPrice(networkId, normalizedTokenId);
+  const { nameAvailable, loading } = useNSMintAvailability(networkId, tokenId);
```

Comment:
all this should be inside the `AvailableNamesInput` component

Reply:
Yep you have to move the logic too in the component, if it's possible, but i think it is